artscii/src/components/main/Main.js

```diff
@@ -32,9 +34,11 @@ function Main() {
             .then(imageUrl => {
                 console.log(`Image URL received in UI - ${imageUrl}`)
                 setImageUrl(imageUrl)
+                setLoading(false);
             })
             .catch(err => {
                 console.log("error encountered = " + err);
+                setLoading(false);
             });
```

Comment:
FWIW you can avoid this duplication with a [`finally`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally):

```suggestion
            })
            .catch(err => {
                console.log("error encountered = " + err);
            })
            .finally(() => {
                setLoading(false);
            });
```

ok love you bye have a nice day

Reply:
https://youtu.be/hvHPv1go75o

Reply:
@contolini  thanks for the suggestion! I updated the PR